docs/advanced/viewports.md

```diff
@@ -30,11 +33,148 @@ Like many things in deck.gl, Viewports can be positioned using a lng/lat "anchor
 
 * Non-pixel projection matrices are obviously bottom left.
-
 * Mercator coordinates are specified in "lng-lat" format [lng, lat, z] format (which naturally corresponds to [x, y, z]).
-
 * Per cartographic tradition, all angles including `latitude`, `longitude`, `pitch` and `bearing` are specified in degrees, not radians.
-
 * Longitude and latitude are specified in degrees from Greenwich meridian and the equator respectively, and altitude is specified in meters above sea level.
 
-* It is possible to query the WebMercatorViewport for a meters per pixel scale. Note that that distance scales are latitude dependent under web mercator projection (see [http://wiki.openstreetmap.org/wiki/Zoom_levels](http://wiki.openstreetmap.org/wiki/Zoom_levels) for more details), so scaling will depend on the viewport center and any linear scale factor should only be expected to be locally correct.
\ No newline at end of file
+It is possible to query the WebMercatorViewport for a meters per pixel scale. Note that that distance scales are latitude dependent under web mercator projection (see [http://wiki.openstreetmap.org/wiki/Zoom_levels](http://wiki.openstreetmap.org/wiki/Zoom_levels) for more details), so scaling will depend on the viewport center and any linear scale factor should only be expected to be locally correct.
+
+
+## Viewport Positioning
+
+Viewports allow the application to specify the position and extent of the viewport (i.e. the target rendering area on the screen). Viewport positions are specified in CSS coordinates (top left, non-retina, these coordinates are different from WebGL coordinates, see remarks below). It is expected that CSS coordinates are most natural to work with, as the rest of the UI layout with other HTML components is done in the CSS coordinate system.
+
+* **x,y coordinates** - Viewports allow specification of x,y coordinates in the viewport in additin to width and height. These are only used for positioning (and not for calculation of intrinsic viewport parameters).
+
+
+## Multiple Viewports
+
+It is common in 3D applications to render a 3D scene multiple times, with different cameras:
+* To show views from multiple viewpoints (cameras), e.g. in a split screen setup.
+* To show a detail view (e.g, first person), and an overlaid, smaller "map" view (e.g. third person or top down, zoomed out to show where the primary viewpoint is).
+* To support stereoscopic rendering, where left and right views are needed, providing the necessary parallax between left and right eye.
+* For rendering into offscreen framebuffers, which can then be used for e.g. advanced visual effects, screen shot solutions, overlays onto DOM elements outside of the primary deck.gl canvas (e.g. a video).
+
+
+### The `DeckGL.viewports` property
+
+Allows the main `DeckGL` component to accept a list of Viewports (and/or viewport descriptors). This will be in addition to, or preferably instead of, the current `viewport` prop.
```

Comment:
> the current `viewport` prop.
There is no `viewport` prop, all the viewport parameters (pitch, bearing, etc) are separate props. Or you will be adding `viewport` prop too?

Reply:
> There is no viewport prop, all the viewport parameters (pitch, bearing, etc) are separate props. Or you will be adding viewport prop too?

Currently we support both direct params and `viewport`, see https://github.com/uber/deck.gl/blob/viewport-docs/src/react/deckgl.js#L48. The viewport was added a while ago to support non-geospatial use cases.